lib/feature/wallet/token_wallet_details/cubit/token_wallet_details_cubit.dart

```diff
@@ -0,0 +1,113 @@
+import 'dart:async';
+
+import 'package:app/app/service/service.dart';
+import 'package:bloc/bloc.dart';
+import 'package:collection/collection.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:nekoton_repository/nekoton_repository.dart';
+
+part 'token_wallet_details_state.dart';
+
+part 'token_wallet_details_cubit.freezed.dart';
+
+/// Cubit that allows loading balance for the [TokenWallet] (history loads
+/// separately) in widget.
+class TokenWalletDetailsCubit extends Cubit<TokenWalletDetailsState> {
+  TokenWalletDetailsCubit({
+    required this.nekotonRepository,
+    required this.rootTokenContract,
+    required this.owner,
+    required this.currencyConvertService,
+    required this.balanceService,
+    required this.assetsService,
+  }) : super(const TokenWalletDetailsState.initial()) {
+    final a = nekotonRepository.seedList.findAccountByAddress(owner);
+    if (a == null) {
+      emit(const TokenWalletDetailsState.empty());
+
+      return;
+    }
+
+    account = a;
+    _cachedFiatBalance = currencyConvertService.convert(Fixed.zero);
+
+    final contract = assetsService.maybeGetTokenContract(
+      rootTokenContract,
+      nekotonRepository.currentTransport,
+    );
+    _cachedTokenBalance = Money.fromBigIntWithCurrency(
+      BigInt.zero,
+      Currency.create(contract?.symbol ?? '-', 0),
```

Comment:
Is it ok to use negative sign as currency symbol?

Reply:
Replaced to `?` as told before